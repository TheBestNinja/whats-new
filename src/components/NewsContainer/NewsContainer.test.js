import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer'

describe('NewsContainer', () => {
  it('should match the snapshot with all the data passed in corretly with a zero or even index', () => {
    // setup
    const news = [
      {
        id: 5,
        img: 'www.101.com',
        headline: 'Hundred and One Dalmatians',
        description: 'Cruella visits their home and expresses contempt upon meeting Roger. Her initial disgust at them having a baby turns to excitement when she finds out Perdy is expecting too.',
        url: 'www.101.com'
      },
      {
        id: 4,
        img: 'www.blackbeautymovie.com',
        headline: 'Black Beauty',
        description: 'This heartwarming adaptation of the novel tells the story of the courageous stallion Black Beauty, a well-bred horse in 19th century England.',
        url: 'www.blackbeautymovie.com'
      }
    ]

    // execution
    const wrapper = shallow(
      <NewsContainer news={news} />
    )

    // expectation
    expect(wrapper).toMatchSnapshot();

  })
})