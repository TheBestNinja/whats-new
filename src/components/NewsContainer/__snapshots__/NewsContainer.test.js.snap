// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the snapshot with all the data passed in corretly with a zero or even index 1`] = `
<div
  className="news-container"
>
  <NewsArticleImgLeft
    article={
      Object {
        "description": "Cruella visits their home and expresses contempt upon meeting Roger. Her initial disgust at them having a baby turns to excitement when she finds out Perdy is expecting too.",
        "headline": "Hundred and One Dalmatians",
        "id": 5,
        "img": "www.101.com",
        "url": "www.101.com",
      }
    }
    key="newsArticle0"
  />
  <NewsArticleImgRight
    article={
      Object {
        "description": "This heartwarming adaptation of the novel tells the story of the courageous stallion Black Beauty, a well-bred horse in 19th century England.",
        "headline": "Black Beauty",
        "id": 4,
        "img": "www.blackbeautymovie.com",
        "url": "www.blackbeautymovie.com",
      }
    }
    key="newsArticle1"
  />
</div>
`;
